v2/apiserver/internal/api/events.go

```diff
@@ -453,14 +457,6 @@ func (e *eventsService) createSingleEvent(
 		}
 	}
 
-	// If no commit (sha) or ref (branch or tag) is specified, default to the
-	// master branch
-	if workerSpec.Git != nil {
-		if workerSpec.Git.Commit == "" && workerSpec.Git.Ref == "" {
-			workerSpec.Git.Ref = "refs/heads/master"
-		}
-	}
-
```

Comment:
Very important-- this means it's now possible for commit _and_ ref to both be empty strings.

The initializer logic is updated accordingly to account for that.